### PR TITLE
Newfood atomization: Burgers & Pizzas & Processable element

### PR DESCRIPTION
**Pull request by Tsar-Salat**
Ports:
- https://github.com/tgstation/tgstation/pull/53224
- https://github.com/tgstation/tgstation/pull/53641
- https://github.com/tgstation/tgstation/pull/54701
- https://github.com/tgstation/tgstation/pull/58983
- https://github.com/tgstation/tgstation/pull/77443

## About The Pull Request
Converts Burgers and Pizza to newfood. Temporarily removes custom Pizza and Burgers.

Adds protein, a nutriment subtype. It can be found in particularly meaty and hearty food, and provides a very minor brute heal.

As pizzas are the first newfood item to properly use the processable element, I cleaned it up, so it made the correct references and grammar.

## Why It's Good For The Game
One more step to newfood.

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

All the pizzas and their slices
pizzasdemonstration

Making and slicing pizza. Correct grammar outputted to chat

https://github.com/BeeStation/BeeStation-Hornet/assets/62388554/a7fea3a0-6634-4f21-b49a-2b8a1aa0250a

Making and slicing burger.

https://github.com/BeeStation/BeeStation-Hornet/assets/62388554/f7698ad7-158a-471a-bbd4-8732fce65eaf

</details>

## Changelog
:cl: RKz, Ghommie, tralezab, cacogen, Krysonism, qustinnus
add: Added Pizza and Burgers to Newfood.
delete: Temporarily removes custom burgers and pizza
tweak: processable element uses the correct grammar and references
code: split pizzaspaghetti.dmi into pizza.dmi and spaghetti.dmi
/:cl:

### CODE REVIEW

File: code/datums/elements/food/processable.dm [hunk 1]
```diff
@@ -40,8 +40,7 @@
 		var/found_table = locate(/obj/structure/table) in found_location
 		var/found_tray = locate(/obj/item/storage/bag/tray) in found_location
 		if(!found_turf && !istype(found_location, /obj/item/storage/bag/tray) || found_turf && !(found_table || found_tray))
-			to_chat(user, "<span class='notice'>You cannot make that here! You need a table or at least a tray.</span>")
+			to_chat(user, "<span class='notice'>You cannot make [initial(result_atom_type.name)] here! You need a table or at least a tray.</span>")
 			return
 
-	mutable_recipes += list(list(TOOL_PROCESSING_RESULT = result_atom_type, TOOL_PROCESSING_AMOUNT = amount_created, TOOL_PROCESSING_TIME = time_to_process))
-	return COMPONENT_NO_AFTERATTACK
```
Comment (itsmeow) [inline code/datums/elements/food/processable.dm:47]: why is this removed?

File: code/game/atoms.dm [hunk 2]
```diff
@@ -1378,16 +1378,30 @@
...
-		qdel(src)
+		//var/list/atom/created_atoms = list() //Customfood
+		var/amount_to_create = chosen_option[TOOL_PROCESSING_AMOUNT]
+		for(var/i = 1 to amount_to_create)
+			var/atom/created_atom = new atom_to_create(drop_location())
+			created_atom.pixel_x = pixel_x
+			created_atom.pixel_y = pixel_y
+			if(i > 1)
+				created_atom.pixel_x += rand(-8,8)
+				created_atom.pixel_y += rand(-8,8)
```
Comment (itsmeow) [inline code/game/atoms.dm:1394]: can we make a proc to do pixel offsets like this? i think its a common code pattern

File: code/modules/awaymissions/corpse.dm [hunk 3]
```diff
@@ -451,7 +451,7 @@
 	name = "Beach Bum"
 	glasses = /obj/item/clothing/glasses/sunglasses
 	r_pocket = /obj/item/storage/wallet/random
-	l_pocket = /obj/item/reagent_containers/food/snacks/pizzaslice/dank;
+	l_pocket = /obj/item/food/pizzaslice/dank;
```
Comment (itsmeow) [inline code/modules/awaymissions/corpse.dm]: semicolon. NUKE IT

**Review by itsmeow — CHANGES_REQUESTED** [review]
Lacks an UpdatePaths

File: _maps/shuttles/emergency/emergency_luxury.dmm [hunk 4]
```diff
@@ -260,7 +260,7 @@
 /area/shuttle/escape/luxury)
 "aX" = (
 /obj/structure/table/wood/fancy,
-/obj/item/reagent_containers/food/snacks/customizable/pizza,
+/obj/item/food/pizza/dank,
```
Comment (itsmeow) [inline _maps/shuttles/emergency/emergency_luxury.dmm:263]: dank pizza cringe

Comment (PestoVerde322) [reply _maps/shuttles/emergency/emergency_luxury.dmm:263]: I think it should be fine for the time being.

File: code/game/atoms.dm (hunk 2 above)
Comment (Tsar-Salat) [reply code/game/atoms.dm:1394]: sure, ill go with a name like `randomize_placement_offsets`

File: code/datums/elements/food/processable.dm (hunk 1 above)
Comment (Tsar-Salat) [reply code/datums/elements/food/processable.dm:47]: me when I undocument changes https://github.com/tgstation/tgstation/pull/58983

Apparentely it fucked shit up on TG, so im removing it after adding it

File: code/modules/awaymissions/corpse.dm (hunk 3 above)
Comment (Tsar-Salat) [reply code/modules/awaymissions/corpse.dm]: fixed in 9e80b25

File: code/game/atoms.dm (hunk 2 above)
Comment (Tsar-Salat) [reply code/game/atoms.dm:1394]: image

procced in 9e80b25

File: _maps/shuttles/emergency/emergency_luxury.dmm (hunk 4 above)
Comment (Tsar-Salat) [reply _maps/shuttles/emergency/emergency_luxury.dmm:263]: > dank pizza cringe

its temporary

**Comment by Tsar-Salat** [comment]
> Lacks an UpdatePaths

Added one `tools/UpdatePaths/Scripts/9860-NEWFOODpizzaburger.txt`

File: code/game/atoms.dm (hunk 2 above)
Comment (itsmeow) [reply code/game/atoms.dm:1394]: this doesnt need to be a proc on /atom/ since you aren't using src

Comment (Tsar-Salat) [reply code/game/atoms.dm:1394]: Yeah, I can't wrap my head around this, and it keeps throwing type info alerts at me after I generic-ized it, however I try and spin it.

Pixels are randomized all over the codebase, and Its out of scope when it only accounts for two lines of code here

**Review by itsmeow — APPROVED** [review]
Looks good to me.
